testbed/testbed/validator.go

```diff
@@ -0,0 +1,312 @@
+			fieldPath:     "SpanId",
+			expectedValue: hex.EncodeToString(sentSpan.SpanId),
+			actualValue:   hex.EncodeToString(recdSpan.SpanId),
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.TraceState != recdSpan.TraceState {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "TraceState",
+			expectedValue: sentSpan.TraceState,
+			actualValue:   recdSpan.TraceState,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if hex.EncodeToString(sentSpan.ParentSpanId) != hex.EncodeToString(recdSpan.ParentSpanId) {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "ParentSpanId",
+			expectedValue: hex.EncodeToString(sentSpan.ParentSpanId),
+			actualValue:   hex.EncodeToString(recdSpan.ParentSpanId),
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.Name != recdSpan.Name {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "Name",
+			expectedValue: sentSpan.Name,
+			actualValue:   recdSpan.Name,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.Kind != recdSpan.Kind {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "Kind",
+			expectedValue: sentSpan.Kind,
+			actualValue:   recdSpan.Kind,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.StartTimeUnixNano != recdSpan.StartTimeUnixNano {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "StartTimeUnixNano",
+			expectedValue: sentSpan.StartTimeUnixNano,
+			actualValue:   recdSpan.StartTimeUnixNano,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.EndTimeUnixNano != recdSpan.EndTimeUnixNano {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "StartTimeUnixNano",
+			expectedValue: sentSpan.EndTimeUnixNano,
+			actualValue:   recdSpan.EndTimeUnixNano,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if len(sentSpan.Attributes) != len(recdSpan.Attributes) {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "Attributes",
+			expectedValue: len(sentSpan.Attributes),
+			actualValue:   len(recdSpan.Attributes),
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.DroppedAttributesCount != recdSpan.DroppedAttributesCount {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "DroppedAttributesCount",
+			expectedValue: sentSpan.DroppedAttributesCount,
+			actualValue:   recdSpan.DroppedAttributesCount,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if len(sentSpan.Events) != len(recdSpan.Events) {
```

Comment:
Same for event values: add TODO to compare.

Reply:
Done